docs/security.md

```diff
@@ -76,6 +76,37 @@ function transferFrom(address who, address to, uint amount)
 }
 ```
 
+## Speed Bump
+
+If we would like to prevent off-chain calls from being chained together, we can
+ensure that the block has been finalized.
+
+```solidity
+contract Secret {
+  uint256 private _height;
+  bytes private _secret;
+  address private _buyer;
+
+  constructor(bytes memory _text) {
+    _secret = _text;
+  }
+
+  function recordPayment() external payable {
+    require(msg.value == 1 ether);
+    // set and lock buyer
+    _height = block.number;
+    _buyer = msg.sender;
+  }
+
+  /// @notice Reveals the secret.
+  function revealSecret() view external returns (bytes memory) {
+    require(block.number > _height, "not settled");
+    require(_buyer != address(0), "no recorded buyer");
```

Comment:
Sorry for more nits, but you could add a comment now saying something like `// TODO: Optionally authenticate buyer's call.` as currently the secret would be revealed to anyone (which is fine to keep this example small, we should just make sure to note that the example is incomplete).

Reply:
Nits welcome! 🫶

Yeah, I think the warning is nice.